Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -42,38 +41,29 @@ extension HTTPClient {
         ///     - defaultDomain: Default domain to use if cookie was sent without one.
         /// - returns: nil if the header is invalid.
         public init?(header: String, defaultDomain: String) {
-            let components = header.components(separatedBy: ";").map {
-                $0.trimmingCharacters(in: .whitespaces)
-            }
+            let components = header.split(separator: ";", omittingEmptySubsequences: false)
```

Comment:
This will perform _vastly_ better if we do this work on the `UTF8View`. Is there any reason we can't?